lib/home_page.dart

```diff
@@ -175,172 +155,220 @@ class HomePageState extends State<HomePage> {
         ),
       ),
     );
-    trips.sort((Trip a, Trip b) => a.startTime.compareTo(b.startTime));
-    _upcomingTripsCount = trips.length;
-    items.insertAll(0, trips);
+    _trips.sort((Trip a, Trip b) => a.startTime.compareTo(b.startTime));
+
+    final List<Map<String, dynamic>> rideEventsData = parseHelper.parseListOfMaps(
+      await supabaseManager.supabaseClient.from('ride_events').select<List<Map<String, dynamic>>>('''
+      *,
+      ride: ride_id(*,
+        rider: rider_id(*),
+        drive: drive_id(*,
+          driver: driver_id(*)
+          )
+      )
+      ''').eq('read', false).order('created_at'),
+    );
+    _rideEvents =
+        RideEvent.fromJsonList(rideEventsData).where((RideEvent rideEvent) => rideEvent.isForCurrentUser()).toList();
+
+    final List<Map<String, dynamic>> messagesData = parseHelper.parseListOfMaps(
+      await supabaseManager.supabaseClient.from('messages').select<List<Map<String, dynamic>>>('''
+      *,
+      sender: sender_id(*)
+      )
+    ''').eq('read', false).neq('sender_id', profileId).order('created_at'),
+    );
+    _messages = Message.fromJsonList(messagesData);
 
     setState(() {
-      _items = items;
       _fullyLoaded = true;
     });
   }
 
   @override
   Widget build(BuildContext context) {
+    final Widget searchButton = Hero(
+      tag: 'SearchButton',
+      transitionOnUserGestures: true,
+      child: Button(
+        S.of(context).pageHomeSearchButton,
+        key: const Key('SearchButton'),
+        onPressed: () => Navigator.of(context).push(
+          MaterialPageRoute<void>(builder: (BuildContext context) => const SearchRidePage()),
+        ),
+      ),
+    );
+    final Widget createButton = Hero(
+      tag: 'CreateButton',
+      transitionOnUserGestures: true,
+      child: Button(
+        S.of(context).pageHomeCreateButton,
+        key: const Key('CreateButton'),
+        onPressed: () => Navigator.of(context).push(
+          MaterialPageRoute<void>(builder: (BuildContext context) => const CreateDrivePage()),
+        ),
+      ),
+    );
+    List<Widget> notifications = <Widget>[];
+    if (_fullyLoaded) {
+      if (_trips.isNotEmpty) {
+        notifications.add(
+          Column(
+            key: const Key('tripsColumn'),
+            children: <Widget>[
+              Align(
+                alignment: Alignment.centerLeft,
+                child: Text(S.of(context).pageHomePageTrips, style: Theme.of(context).textTheme.titleLarge),
+              ),
+              const SizedBox(height: 10),
```

Comment:
can we have a little more vertical space... the categories are sticking together a little too much for my taste